### PR TITLE
Made changes needed by expo-payments-stripe unimodule

### PR DESCRIPTION
**Pull request by Szymon20000**

### CODE REVIEW

File: packages/xdl/src/detach/AndroidShellApp.js [hunk 1]
```diff
@@ -335,6 +335,16 @@ exports.createAndroidShellAppAsync = async function createAndroidShellAppAsync(a
   }
 };
 
+function createUUID() {
```
Comment (dsokal) [inline packages/xdl/src/detach/AndroidShellApp.js]: ```import uuid from 'uuid';```?

File: packages/xdl/src/detach/AndroidShellApp.js [hunk 2]
```diff
@@ -692,6 +702,23 @@ export async function runShellAppModificationsAsync(
     );
   }
 
+  // Change stripe schemes and add meta-data
+  const randomID = createUUID();
+  const newScheme =
+    '<meta-data android:name="standaloneStripeScheme" android:value=".' + randomID + '" />';
```
Comment (dsokal) [inline packages/xdl/src/detach/AndroidShellApp.js]: template string

File: packages/xdl/src/detach/AndroidShellApp.js [hunk 3]
```diff
@@ -692,6 +702,23 @@ export async function runShellAppModificationsAsync(
...
+  await regexFileAsync(
+    '<!-- ADD HERE STRIPE SCHEME META DATA -->',
+    newScheme,
+    path.join(shellPath, 'app', 'src', 'main', 'AndroidManifest.xml')
+  );
+
+  const newSchemeSuffix = 'expo.modules.payments.stripe.' + randomID + '" />';
```
Comment (dsokal) [inline packages/xdl/src/detach/AndroidShellApp.js]: template string

**Comment by fson** [comment]
Note that before this can be merged, it needs to work with all supported SDK versions, starting from 23.

I believe this might require adding some metadata to unimodule configuration that says which semver range each unimodule is available for?

**Pull request merged**